src/mlpack/methods/ann/layer/padding_impl.hpp

```diff
@@ -35,7 +35,7 @@ PaddingType<MatType>::PaddingType(
 }
 
 template<typename MatType>
-PaddingType<MatType>::PaddingType(const PaddingType& other) :
+PaddingType<MatType>::PaddingType(const PaddingType<MatType>& other) :
```

Comment:
Seems to be fine.  My bad.  I'm not sure now what prompted me to add these in the first place -- I thought I added them because they were causing issues for me (I am using MatType=arma::fmat, so not the default).
I will remove.